### DISCUSSION TITLE
💥 Project Vision 💥

### DISCUSSION THREAD

**Discussion by Hona** [Hona/TempusHub2#16] [Announcements]
Login with Steam.

✨ Planned Features:

- Player Dashboard
  - Suggested Maps for Completion, or TT (based on other 'similar' players completion/tt which you should be similar to) #21 
  - Fun/Competitive maps suggested based on player ratings #21 
  - Easy WR/Easy TT based on similar players (e.g. you have 95% match but don't have TT on something they have WR) #21 
  - Friend Activity (steam friends?, site friends, site following?) #22 
  - Progress Bars: Completion. TTs, WRs  #23 
  - Map List Summary (bar goes from 0 (not complete) to 100% has WR). Sort by best/worst maps. #23 
  - Favorited/Pinned maps #25 
  - player WR dominance pie chart #20 
  - Spotify recapped for each player (most played map, tts gained, wrs lost blah blah) #22 
  
- Map View
  - Leaderboards/Points (EV, expected value, e.g. 50th percentile time vs points, or, similar players to you completion median time vs points) #20 
  - Comments: Text, videos whatever. think maze tips, dumb setups, comments, how to cheat no tele.  #25 
  - Community Data:
    - Images/Videos for the gallery #25 
    - Rating (difficulty + fun - separately, verified using completion) #25 
    - Tags (airpogo, hardpogo, speedshot etc. Should tags be community sourced?) #25 
  - Auto Videos - any leaderboard run that was/is WR links to Tempus Records/Archive magically (no more searching youtube slowly) #26 

- General Community sourced data:
  - peoples autoexec & config #25 
  - scripts #25 
  - custom textures/particles etc (all the hidden stuff that you have to ask to know about) #25 

- Analytics: (we can do lots of cool stuff)
  - Rank history. Track all player ranks. Fastest growing histograph... one day animation after enough data. #20 
  - Map WR history. animation also #20 
  both these history things you see as youtube videos
  - What if (allow custom rank formula, map tiers?) #30 
  - Time spent on Maps #29 
  - Most run map/bonus  #29 
  - Most avoided  #29 
  - Most WRd  #20 
  - Biggest gap between run # 1-> # 2 and/or # 1- # 10  #20 
  - Rank changes (e.g. Hona just ranked up from 12 -> 11) #20 

- Mobile App/PWA/Installed (even safari allows native push notifications lol) 
  - WR notifications (allow all maps or only watched/favourited/followed maps or players) #24 
  - Player on given Map #24 
  - You lost a TT #24 
  - You lost a WR #24 
  - a # 2 or any top player/TT holder playing a map you have WR on #24 

- Querying
  - Capture historical data for permanent access. Allow querying ranges, specific dates & generally expose what data WAS. #20 

- 💣 Jump as a Service
  - Tempus Records/Archive on demand: Serverless videos, e.g. request any run/player, WR zone completion whatever and get a video back. [Azure VM GPU](https://learn.microsoft.com/en-us/azure/virtual-machines/sizes-gpu)  #30 
    - Lets say you do a completion on a hard t6 for the first time, and want to uplaod it. Most people do it wrong or can't be bothered. Click a button then get a mp4 download using Tempus Records config/quality
    Spin up a VM then record video & dispose the VM. That way I can see how much $$$ im burning on cloud compute  #30 
  - Tempus TV on demand. (potentially many instances)  #30 

- STV Viewer
  - Custom upload? #28 
  - always on for WRs, map runs #19 
  - Downlaod then display the chat logs to users. Any WRs etc.  #19 

- Clans
  - steam group? nickname suffix? #27 
  - Clan leaderboards #27 
  - Clan war. (set period of time, who gets the most points/WRs/TTs/Completion) #27 
---

**Notes from Jumpers:**

Boshy: 
  - Important: Make it easy for new jumpers to get into it #25 
  - Images/gallery for jump more than the 1 thumbnail #25 

Luke: 
  - Content moderation #25

**Comment by MEGAWOP** [comment]
Seems good give now
